app/ui/legacy/src/main/java/com/fsck/k9/activity/MessageList.kt

```diff
@@ -738,101 +738,126 @@ open class MessageList :
                     return true
                 }
             }
-            KeyEvent.KEYCODE_C -> {
+            KeyEvent.KEYCODE_DEL -> {
+                if (displayMode == DisplayMode.MESSAGE_LIST) {
+                    messageListFragment!!.onDelete()
+                } else if (messageViewFragment != null) {
+                    messageViewFragment!!.onDelete()
+                }
```

Comment:
Can you please extract these lines to a new method named `onDeleteHotkey` or similar, and then also call this method when handling the 'd' key below. Otherwise it's too easy to miss a copy when someone wants to change how the app reacts to the delete hotkey.